application/frontend/src/app/core/selectors/timeline.selectors.ts

```diff
@@ -112,9 +112,18 @@ const getTimeline = (
     'Visits and travel steps not aligned.'
   );
 
-  const breaks = route.breaks ? route.breaks.map(createBreakSegment) : [];
   const vehicleStartTime = route.vehicleStartTime ? durationSeconds(route.vehicleStartTime) : null;
   const vehicleEndTime = route.vehicleEndTime ? durationSeconds(route.vehicleEndTime) : null;
+  // Create break segments, excluding those that are outside of the vehicle's day
+  const breaks = route.breaks
+    ? route.breaks
+        .map(createBreakSegment)
+        .filter(
+          (brk) =>
+            brk.startTime.greaterThanOrEqual(vehicleStartTime) &&
```

Comment:
What would we expect the behavior to be if `vehicleStartTime` and/or `vehicleEndTime` are `null`? In those cases, this code would remove all break segments, which seems like it could the desired behavior (i.e. can't schedule a break if you didn't schedule start/end time). Mostly asking if it's valid to send a scenario to CFR that includes breaks without also setting vehicle start/end.

Reply:
In this case, `vehicleStartTime` and `vehicleEndTime` are the start and end times of the route and aren't related to the vehicle time windows. May be worth renaming to clarify that, but the question still stands of whether or not a route  would ever have a `null` start or end? I'd assume not, but need to double check.

Reply:
Renamed `vehicleStartTime` and `vehicleEndTime` to `routeStartTime` and `routeEndTime` to make it clearer what we're checking against.